sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -104,7 +105,13 @@ case class DataSourceAnalysis(conf: SQLConf) extends Rule[LogicalPlan] with Cast
         None
       } else if (potentialSpecs.size == 1) {
         val partValue = potentialSpecs.head._2
-        Some(Alias(cast(Literal(partValue), field.dataType), field.name)())
+        conf.storeAssignmentPolicy match {
+          case StoreAssignmentPolicy.ANSI | StoreAssignmentPolicy.STRICT =>
```

Comment:
the major difference between ANSI and STRICT mode is: STRICT mode fails at analysis phase. Can we follow the behavior in `PreprocessTableInsertion`?

Reply:
The parser erases the type info of static partition values and converts them to string, so we can't really do a compile-time type check here...

Reply:
Since we lost the data type of static partition values in parser, it seems hard for us to follow `canWrite`  in `PreprocessTableInsertion`(because we can not figure out `writeType`).

Reply:
Probably, better to leave some comments about why this check is placed here for making other developers understood easily.

Reply:
yea let's add some comments

Reply:
Added, thanks.